### ISSUE TITLE
libLLVM-14git.so not in library search paths

### ISSUE THREAD

**Issue by knzivid** [rocm-arch/rocm-arch#705]
Exporting `LD_LIBRARY_PATH=/opt/rocm/llvm/lib` is needed to make `clinfo -l` show my RX 5500. Here are my installed packages

```
rocm-cmake 5.0.2-1
rocm-core 5.0.1-1
rocm-device-libs 5.0.2-1
rocm-hip-runtime 5.0.2-1
rocm-language-runtime 5.0.2-1
rocm-llvm 5.0.2-2
rocm-opencl-runtime 5.0.2-1
rocm-opencl-sdk 5.0.2-1
rocm-smi-lib 5.0.2-1
rocminfo 5.0.2-1
```

Is `/opt/rocm/llvm/lib` added to `/etc/ld.so.conf.d/` by any other rocm family packages?

**Comment by tpkessler** [comment]
`rocm-llvm` should not be needed for `clinfo` to work. Can you please post the output of `strace /opt/rocm/bin/clinfo -l 2>&1 | grep llvm` and `strace clinfo -l 2>&1 | grep llvm`?

**Comment by knzivid** [comment]
Earlier, I actually discovered I needed `/opt/rocm/llvm/lib/libLLVM-14git.so` by looking at the strace output. 

I no longer have the AMD GPU with me. But I don't think that matters for the library loader. Here's the output from both commands

```
$ strace -f /opt/rocm/bin/clinfo -l 2>&1 | grep llvm 
[pid 87435] openat(AT_FDCWD, "/opt/rocm/llvm/lib/libLLVM-14git.so", O_RDONLY|O_CLOEXEC) = 11
$ strace -f clinfo -l 2>&1 | grep llvm 
[pid 87468] openat(AT_FDCWD, "/opt/rocm/llvm/lib/libLLVM-14git.so", O_RDONLY|O_CLOEXEC) = 11
```

Here's a tiny snippet from `LD_DEBUG=bindings clinfo`

```
87552:     binding file /opt/rocm/lib/libamd_comgr.so.2 [0] to /usr/lib/libc.so.6 [0]: normal symbol `sched_yield' [GLIBC_2.2.5]
87552:     binding file /opt/rocm/lib/libamd_comgr.so.2 [0] to /opt/rocm/llvm/lib/libLLVM-14git.so [0]: normal symbol `_ZN4llvm15LockFileManager13waitForUnlockEj' [LLVM_14]
87552:     binding file /opt/rocm/lib/libamd_comgr.so.2 [0] to /opt/rocm/llvm/lib/libLLVM-14git.so [0]: normal symbol `_ZN4llvm24DisableABIBreakingChecksE' [LLVM_14]
87552:     binding file /opt/rocm/lib/libamd_comgr.so.2 [0] to /usr/lib/libstdc++.so.6 [0]: normal symbol `__gxx_personality_v0' [CXXABI_1.3]
87552:
87552:     calling init: /opt/rocm/llvm/lib/libLLVM-14git.so
87552:
87552:
87552:     calling init: /opt/rocm/lib/libamd_comgr.so.2
```

**Comment by tpkessler** [comment]
On my machine:
```bash
strace -f /opt/rocm/bin/clinfo -l 2>&1 | grep llvm

echo $?
1
```
(no match) and for
```bash
LD_DEBUG=bindings clinfo 2>&1 | grep llvm

echo $?
1
```

**Comment by PatriceBlin** [comment]
`comgr` (5.0.2-1) do look for `libLLVM-14git.so`

```console
% readelf -d /opt/rocm/lib/libamd_comgr.so

Dynamic section at offset 0x3c7c0b8 contains 30 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libLLVM-14git.so]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [ld-linux-x86-64.so.2]
 0x000000000000000e (SONAME)             Library soname: [libamd_comgr.so.2]
 [...]
 ```

I had the same issue as @knzivid, but not with clinfo.
For me clinfo works fine but using AMD's tensorflow I have `HIP_ERROR_NoDevice` (while clinfo detect the GPU)

```python
import tensorflow as tf

print("Num GPUs Available: ",
        len(tf.config.experimental.list_physical_devices('GPU')))
```

```console
$ python /tmp/test_gpu.py
2022-03-31 11:33:28.328990: E tensorflow/stream_executor/rocm/rocm_driver.cc:993] could not retrieve ROCM device count: HIP_ERROR_NoDevice
Num GPUs Available:  0
```

**Comment by wsippel** [comment]
This also causes the Blender experimental build with HIP support to fail. Blender initializes HIP, HIP tries to open libamd_comgr.so, which fails because it can't find libLLVM-14git.so. Running Blender with $LD_LIBRARY_PATH set to /opt/rocm/llvm/lib works.

**Comment by tpkessler** [comment]
Should be fixed with 9d0d918d71638c035da0a8b0980e5c459e7d1e60

**Comment by acxz** [comment]
Going ahead and closing this issue. Please comment on this issue if you are still experiencing issues!
